lib/msal-browser/test/interaction_client/RedirectClient.spec.ts

```diff
@@ -1026,6 +1026,60 @@ describe("RedirectClient", () => {
             expect(loginUrlSpy.calledWith(validatedRequest)).toBeTruthy();
         });
 
+        it("Uses adal token from cache if it is present and sets preferred_name as the login hint.", async () => {
```

Comment:
Can you also add another test where `idTokenClaims` contains both a upn (with the "mangled" email value) and preferred_username, and shows that preferred_username is used?